airbyte-integrations/connectors/source-surveymonkey/source_surveymonkey/streams.py

```diff
@@ -0,0 +1,257 @@
+#
+# MIT License
+#
+# Copyright (c) 2020 Airbyte
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+#
+
+import tempfile
+import urllib.parse
+from abc import ABC, abstractmethod
+from typing import Any, Iterable, List, Mapping, MutableMapping, Optional
+
+import pendulum
+import requests
+import vcr
+from airbyte_cdk.models import SyncMode
+from airbyte_cdk.sources.streams.http import HttpStream
+
+cache_file = tempfile.NamedTemporaryFile()
+
+
+class SurveymonkeyStream(HttpStream, ABC):
```

Comment:
this stream should handle the rate limits described here: https://developer.surveymonkey.com/api/v3/#request-and-response-limits

Reply:
I have added the backoff but commented daily resolve. Don't sure if I need to uncomment it: daily limit is very low, it can be easily broken by a user with a data unless he has paid unlimited api access. The delay may be up to 23+ hours per 500 requests, so every single sync may have a several days. Please tell me if I need to uncomment this  and this will be good if we can have so huge delays...

https://github.com/airbytehq/airbyte/blob/f831465659b4a9881ec9fcdcf515574e96c42262/airbyte-integrations/connectors/source-surveymonkey/source_surveymonkey/streams.py#L49

Reply:
Also can you create another issue to Make the Surveymonkey connector using Oauth2.0 via a Public app to avoid these rate limits? public apps don't have the rate limits. Once we have Airbyte cloud we can use a single centralized app available on surveymonkey's appstore.